Traitsui integrationtests fail when used with pyface from master branch.

The following two test failures occur when the traitsui testsuite is run using pyface from source (discovered in PR https://github.com/enthought/traitsui/pull/1427).

<details>

```
======================================================================
FAIL: test_run (test_all_examples.TestExample) (file_path='C:\\Users\\rporuri\\.edm\\envs\\traitsui-test-3.6-pyqt5\\lib\\site-packages\\traitsui\\examples\\demo\\Applications\\Python_source_browser.py')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 332, in test_run
    run_file(file_path)
TypeError: '__main__' is not a package

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 339, in test_run
    file_path, exc, message
AssertionError: Executing C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\examples\demo\Applications\Python_source_browser.py failed with exception '__main__' is not a package
 Traceback (most recent call last):
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 332, in test_run
    run_file(file_path)
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 273, in run_file
    exec(content, globals)
  File "<string>", line 176, in <module>
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 232, in replaced_configure_traits
    with UITester().create_ui(instance, ui_kwargs):
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\testing\tester\ui_tester.py", line 88, in create_ui
    ui = object.edit_traits(**ui_kwargs)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traits\has_traits.py", line 1744, in edit_traits
    args,
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\view.py", line 462, in ui
    ui.ui(parent, kind)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\ui.py", line 244, in ui
    self.rebuild(self, parent)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\toolkit.py", line 163, in ui_live
    ui_live.ui_live(ui, parent)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_live.py", line 43, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_live.py", line 65, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_base.py", line 286, in display_ui
    ui.owner.init(ui, parent, style)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_live.py", line 225, in init
    self.add_contents(panel(ui), bbox)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 259, in panel
    panel = _GroupPanel(content[0], ui).control
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 555, in __init__
    self._add_splitter_items(content, splitter)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 648, in _add_splitter_items
    panel = _GroupPanel(item, self.ui, suppress_label=True).control
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 555, in __init__
    self._add_splitter_items(content, splitter)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 650, in _add_splitter_items
    panel = self._add_items([item])
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 891, in _add_items
    editor.prepare(inner)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\editor.py", line 236, in prepare
    self.init(parent)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\tabular_editor.py", line 240, in init
    self._add_image(image_resource)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\tabular_editor.py", line 351, in _add_image
    image = image_resource.create_icon()
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\ui\qt4\image_resource.py", line 55, in create_icon
    ref = self._get_ref(size)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\i_image_resource.py", line 194, in _get_ref
    self.name, self.search_path, size
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\resource\resource_manager.py", line 67, in locate_image
    return self._locate_image(image_name, resource_path, size)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\resource\resource_manager.py", line 123, in _locate_image
    files(dirname.__name__)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\importlib_resources\_common.py", line 24, in files
    return from_package(get_package(package))
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\importlib_resources\_common.py", line 73, in get_package
    raise TypeError('{!r} is not a package'.format(package))
TypeError: '__main__' is not a package


======================================================================
FAIL: test_run (test_all_examples.TestExample) (file_path='C:\\Users\\rporuri\\.edm\\envs\\traitsui-test-3.6-pyqt5\\lib\\site-packages\\traitsui\\examples\\demo\\Extras\\Image_editor_demo.py')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 332, in test_run
    run_file(file_path)
TypeError: '__main__' is not a package

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 339, in test_run
    file_path, exc, message
AssertionError: Executing C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\examples\demo\Extras\Image_editor_demo.py failed with exception '__main__' is not a package
 Traceback (most recent call last):
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 332, in test_run
    run_file(file_path)
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 273, in run_file
    exec(content, globals)
  File "<string>", line 77, in <module>
  File "C:\Users\rporuri\work\github\ets\traitsui\integrationtests\test_all_examples.py", line 232, in replaced_configure_traits
    with UITester().create_ui(instance, ui_kwargs):
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\testing\tester\ui_tester.py", line 88, in create_ui
    ui = object.edit_traits(**ui_kwargs)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traits\has_traits.py", line 1744, in edit_traits
    args,
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\view.py", line 462, in ui
    ui.ui(parent, kind)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\ui.py", line 244, in ui
    self.rebuild(self, parent)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\toolkit.py", line 163, in ui_live
    ui_live.ui_live(ui, parent)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_live.py", line 43, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_live.py", line 65, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_base.py", line 286, in display_ui
    ui.owner.init(ui, parent, style)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_live.py", line 225, in init
    self.add_contents(panel(ui), bbox)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 259, in panel
    panel = _GroupPanel(content[0], ui).control
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 615, in __init__
    layout = self._add_groups(content, inner)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 696, in _add_groups
    panel = _GroupPanel(subgroup, self.ui).control
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 617, in __init__
    layout = self._add_items(content, inner)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\ui_panel.py", line 891, in _add_items
    editor.prepare(inner)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\editor.py", line 236, in prepare
    self.init(parent)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\traitsui\qt4\image_editor.py", line 278, in init
    self.control.setPixmap(convert_bitmap(image))
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\ui_traits.py", line 85, in convert_bitmap
    ] = bitmap = image_resource.create_bitmap()
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\i_image_resource.py", line 164, in create_image
    ref = self._get_ref(size)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\i_image_resource.py", line 194, in _get_ref
    self.name, self.search_path, size
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\resource\resource_manager.py", line 67, in locate_image
    return self._locate_image(image_name, resource_path, size)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\pyface\resource\resource_manager.py", line 123, in _locate_image
    files(dirname.__name__)
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\importlib_resources\_common.py", line 24, in files
    return from_package(get_package(package))
  File "C:\Users\rporuri\.edm\envs\traitsui-test-3.6-pyqt5\lib\site-packages\importlib_resources\_common.py", line 73, in get_package
    raise TypeError('{!r} is not a package'.format(package))
TypeError: '__main__' is not a package

```

</details>

This seems to be related to the recent change from `pkg_resources` to `importlib_metadata` in #790 .

--- a/pyface/resource/resource_manager.py
+++ b/pyface/resource/resource_manager.py
@@ -50,7 +50,28 @@ class ResourceManager(HasTraits):
     # ------------------------------------------------------------------------
 
     def locate_image(self, image_name, path, size=None):
-        """ Locates an image. """
+        """ Locates an image.
+
+        Parameters
+        ----------
+        image_name : str
+            Name of the image file.
+        path : list of (str or ModuleType)
+            Paths from which image files will be searched. Note that for each
+            path, a subdirectory named 'images' will be search first.
+            The first match will be returned.
+        size : tuple of (m: int, n: int), optional
+            Specific size of the image requested. If provided, then
+            the subdirectory ``images/{m}x{n}`` will be searched first,
+            followed by the ``images`` subdirectory and its containing folder.
+            Default is None.
+
+        Returns
+        -------
+        image_ref : ImageReference or None
+            ImageReference to the image found, or None if no matching images
+            are found.
+        """
 
         if not isinstance(path, collections.abc.Sequence):
             path = [path]
@@ -110,27 +131,24 @@ class ResourceManager(HasTraits):
         else:
             subdirs = ["images/%dx%d" % (size[0], size[1]), "images", ""]
 
+        # Concrete image filenames to be searched
+        image_filenames = [basename + extension for extension in extensions]
+
         for dirname in resource_path:
 
             # If we come across a reference to a module, try and find the
             # image inside of an .egg, .zip, etc.
             if isinstance(dirname, types.ModuleType):
-                for path in subdirs:
-                    for extension in extensions:
-                        searchpath = "%s/%s%s" % (path, basename, extension)
-                        try:
-                            data = (
-                                files(dirname.__name__)
-                                .joinpath(searchpath)
-                                .read_bytes()
-                            )
-                            return ImageReference(
-                                self.resource_factory, data=data
-                            )
-                        except IOError:
-                            pass
-                else:
+                try:
+                    data = _find_resource_data(
+                        dirname, subdirs, image_filenames
+                    )
+                except IOError:
                     continue
+                else:
+                    return ImageReference(
+                        self.resource_factory, data=data
+                    )
 
             # Is there anything resembling the image name in the directory?
             for path in subdirs:
@@ -238,3 +256,80 @@ class ResourceManager(HasTraits):
                 break
 
         return resource_path
+
+
+def _get_package_data(module, rel_path):
+    """ Return package data in bytes for the given module and resource path.
+
+    Parameters
+    ----------
+    module : ModuleType
+        A module from which package data will be discovered.
+        If the module name does not conform to the package requirement, then
+        its "__file__" attribute is used for locating the directory to search
+        for resource files.
+    rel_path : str
+        "/"-separated path for loading data file.
+
+    Returns
+    -------
+    data : bytes
+        Loaded data in bytes.
+
+    Raises
+    ------
+    OSError
+        If the path referenced does not resolve to an existing file or the
+        file cannot be read.
+    """
+
+    if (module.__spec__ is None
+            or module.__spec__.submodule_search_locations is None):
+        module_dir_path = os.path.dirname(module.__file__)
+        path = os.path.join(module_dir_path, *rel_path.split("/"))
+        with open(path, "rb") as fp:
+            return fp.read()
+
+    return (
+        files(module.__name__).joinpath(rel_path).read_bytes()
+    )
+
+
+def _find_resource_data(module, subdirs, filenames):
+    """ For the given module, search directories and names, find a matching
+    resource file and return its content as bytes.
+
+    Parameters
+    ----------
+    module : ModuleType
+        A module from which package data will be discovered.
+        If the module name does not conform to the package requirement, then
+        its "__file__" attribute is used for locating the directory to search
+        for resource files.
+    subdirs : list of str
+        Name of subdirectories to try. Each value can be a "/"-separated
+        string to represent more nested subdirectories.
+    filenames : list of str
+        File names to try.
+
+    Returns
+    -------
+    data : bytes
+        Loaded data in bytes.
+
+    Raises
+    ------
+    OSError
+        If the path referenced does not resolve to an existing file or the
+        file cannot be read.
+    """
+    for path in subdirs:
+        for filename in filenames:
+            searchpath = "%s/%s" % (path, filename)
+            try:
+                return _get_package_data(module, searchpath)
+            except OSError:
+                pass
+    raise OSError(
+        "Unable to load data for the given module and search paths."
+    )

--- a/pyface/resource/resource_manager.py
+++ b/pyface/resource/resource_manager.py
@@ -291,7 +291,7 @@ def _get_package_data(module, rel_path):
             return fp.read()
 
     return (
-        files(module.__name__).joinpath(rel_path).read_bytes()
+        files(module).joinpath(rel_path).read_bytes()
     )
 
 

--- a/pyface/resource/resource_manager.py
+++ b/pyface/resource/resource_manager.py
@@ -143,7 +143,7 @@ class ResourceManager(HasTraits):
                     data = _find_resource_data(
                         dirname, subdirs, image_filenames
                     )
-                except IOError:
+                except OSError:
                     continue
                 else:
                     return ImageReference(

--- a/pyface/tests/test_resource_manager.py
+++ b/pyface/tests/test_resource_manager.py
@@ -53,8 +53,8 @@ class TestPyfaceResourceFactory(unittest.TestCase):
         self.assertEqual(IMAGE_PATH, img_ref.filename)
 
     def test_locate_image_with_module(self):
-        # images/close.png is included in pyface package data.
-        # ResourceManager can find it via package resources
+        # ResourceManager should be able to find the images/close.png which
+        # is included in pyface package data.
         resource_manager = ResourceManager()
         image_ref = resource_manager.locate_image("close.png", [pyface])
         self.assertGreater(len(image_ref.data), 0)

--- a/pyface/tests/test_resource_manager.py
+++ b/pyface/tests/test_resource_manager.py
@@ -68,11 +68,12 @@ class TestPyfaceResourceFactory(unittest.TestCase):
         self.assertIsNone(image_ref)
 
     def test_locate_image_with_name_being_dunder_main(self):
-        # When a module is __main__, we will fall back to use __file__
+        # When a module is not a package, we will fall back to use __file__
 
         # given a module from which there is an image in the same folder
         with tempfile.TemporaryDirectory() as tmp_dir:
             shutil.copyfile(IMAGE_PATH, os.path.join(tmp_dir, "random.png"))
+            # create an empty file for creating a module.
             py_filepath = os.path.join(tmp_dir, "tmp.py")
             with open(py_filepath, "w", encoding="utf-8"):
                 pass

--- a/pyface/tests/test_resource_manager.py
+++ b/pyface/tests/test_resource_manager.py
@@ -10,9 +10,13 @@
 
 
 from collections.abc import Sequence
+import importlib.util
 import os
+import shutil
+import tempfile
 import unittest
 
+import pyface     # a package with images as package resources
 from ..resource_manager import PyfaceResourceFactory
 from ..resource_manager import ResourceManager
 
@@ -47,3 +51,41 @@ class TestPyfaceResourceFactory(unittest.TestCase):
         resource_manager = ResourceManager()
         img_ref = resource_manager.locate_image("core.png", sequence)
         self.assertEqual(IMAGE_PATH, img_ref.filename)
+
+    def test_locate_image_with_module(self):
+        # images/close.png is included in pyface package data.
+        # ResourceManager can find it via package resources
+        resource_manager = ResourceManager()
+        image_ref = resource_manager.locate_image("close.png", [pyface])
+        self.assertGreater(len(image_ref.data), 0)
+
+    def test_locate_image_with_module_missing_file(self):
+        # The required image is not found, locate_image should return None.
+        resource_manager = ResourceManager()
+        image_ref = resource_manager.locate_image(
+            "does_not_exist.png", [pyface]
+        )
+        self.assertIsNone(image_ref)
+
+    def test_locate_image_with_name_being_dunder_main(self):
+        # When a module is __main__, we will fall back to use __file__
+
+        # given a module from which there is an image in the same folder
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            shutil.copyfile(IMAGE_PATH, os.path.join(tmp_dir, "random.png"))
+            py_filepath = os.path.join(tmp_dir, "tmp.py")
+            with open(py_filepath, "w", encoding="utf-8"):
+                pass
+            spec = importlib.util.spec_from_file_location(
+                "__main__", py_filepath
+            )
+            module = importlib.util.module_from_spec(spec)
+
+            resource_manager = ResourceManager(
+                resource_factory=PyfaceResourceFactory()
+            )
+            # when
+            image_ref = resource_manager.load_image("random.png", [module])
+
+            # then
+            self.assertIsNotNone(image_ref)

